imgui-SFML.cpp

```diff
@@ -529,45 +529,39 @@ void Render() {
 }
 
 void Shutdown(const sf::Window& window) {
+    if (s_windowContexts.find(window.getSystemHandle()) == s_windowContexts.end()) {
+        assert(false &&
+                "Window wasn't inited properly: forgot to call ImGui::SFML::Init(window)?");
+    }
+
+    // check if the current window's context is the one we want to remove
+    const bool isCurrentWindowCtx =
+        window.getSystemHandle() == s_currWindowCtx->window->getSystemHandle();
+
+    // remove window's context  
+    delete s_windowContexts[window.getSystemHandle()];
+    s_windowContexts.erase(window.getSystemHandle());
+
     // set current context to some window for convenience if needed
-    if (window.getSystemHandle() == s_currWindowCtx->window->getSystemHandle()) {
-        if (s_windowContexts.size() > 1) {
+    if (isCurrentWindowCtx) {
+        if (!s_windowContexts.empty()) {
             // set to some other window
-            for (std::size_t i = 0; i < s_windowContexts.size(); ++i) {
-                if (s_windowContexts[i]->window->getSystemHandle() != window.getSystemHandle()) {
-                    s_currWindowCtx = s_windowContexts[i];
-                    ImGui::SetCurrentContext(s_currWindowCtx->imContext);
-                    break;
-                }
-            }
+            s_currWindowCtx = s_windowContexts.begin()->second;
+            ImGui::SetCurrentContext(s_currWindowCtx->imContext);
         } else {
             // no alternatives...
             s_currWindowCtx = NULL;
+            ImGui::SetCurrentContext(NULL);
```

Comment:
It seems that this `SetCurrentContext(NULL)` is a change in behavior, previously there wasn't such an action.

Reply:
Not just `unordered_map`, but `std::unique_ptr` should be used as well, probably. :)